src/main/webapp/app/lecture/pdf-preview/pdf-preview.component.ts

```diff
@@ -0,0 +1,203 @@
+import { Component, ElementRef, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { AttachmentService } from 'app/lecture/attachment.service';
+import * as PDFJS from 'pdfjs-dist';
+import 'pdfjs-dist/build/pdf.worker';
+import { Attachment } from 'app/entities/attachment.model';
+import { AttachmentUnit } from 'app/entities/lecture-unit/attachmentUnit.model';
+import { AttachmentUnitService } from 'app/lecture/lecture-unit/lecture-unit-management/attachmentUnit.service';
+import { onError } from 'app/shared/util/global.utils';
+import { AlertService } from 'app/core/util/alert.service';
+
+@Component({
+    selector: 'jhi-pdf-preview-component',
+    templateUrl: './pdf-preview.component.html',
+    styleUrls: ['./pdf-preview.component.scss'],
+})
+export class PdfPreviewComponent implements OnInit, OnDestroy {
+    @ViewChild('pdfContainer', { static: true }) pdfContainer: ElementRef;
+    @ViewChild('enlargedCanvas') enlargedCanvas: ElementRef;
+    attachment?: Attachment;
+    attachmentUnit?: AttachmentUnit;
+    isEnlargedView: boolean = false;
+    currentPage: number = 1;
+    totalPages: number = 0;
+
+    constructor(
+        private route: ActivatedRoute,
+        private attachmentService: AttachmentService,
+        private attachmentUnitService: AttachmentUnitService,
+        private alertService: AlertService,
+    ) {}
+
+    ngOnInit() {
+        this.route.data.subscribe((data) => {
```

Comment:
I am not sure if angular handles the subscription properly here, as far as I am ware angular suggests to unsubscribe to prevent memory leaks

Reply:
Added unsubscribe methods